### PR TITLE
add assign_attrs method

### LINKED ISSUE #1281: update_attrs method
Updating attrs is an operation that consistently breaks up my method chaining. This should be easy to solve with something like

````
def update_attrs(self, *args, **kwargs):
    self.attrs.update(*args, **kwargs)
    return self
````

Thoughts?

edit: I guess it should return a modified object rather than acting as an in-place operation...is copying appropriate here? How is this usually handled?

### PR DESCRIPTION
**Pull request by hsharrison**
- [x] closes #1281 
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff`` (except for whatsnew... wasn't sure there)
 - [x] whatsnew entry

### CODE REVIEW

File: xarray/core/common.py [hunk 1]
```diff
@@ -346,6 +346,28 @@ def assign_coords(self, **kwargs):
         data.coords.update(results)
         return data
 
+    def assign_attrs(self, *args, **kwargs):
+        """Assign new attrs to this object, returning a new object equivalent
```
Comment (shoyer) [inline xarray/core/common.py]: Can you move the second half of this to a new line? We try to keep the title to one line.

Comment (hsharrison) [inline xarray/core/common.py]: I'm sorry, I don't follow. I tried to match the conventions of the rest of the file. Do you mean this:

```
def assign_attrs(self, *args, **kwargs):
    """
    Assign new attrs to this object, returning a new object equivalent
    to...
```

or

```
def assign_attrs(self, *args, **kwargs):
    """Assign new attrs to this object.

    Returns a new object equivalent to...
```

Comment (shoyer) [inline xarray/core/common.py]: The second version is preferred. But you're right that `assign_coords` also has a docstring that violates this.

Comment (shoyer) [inline xarray/core/common.py]: If you like, it would be awesome to clean that up, too.

**Comment by shoyer** [comment]
Awesome, thank you!

**Pull request merged**